src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/study/StudyAdherenceReportGenerator.java

```diff
@@ -0,0 +1,272 @@
+            nextActivity = getNextActivity(weeks, todayLocal);
+        }
+        DateRange dateRange = null;
+        if (eventReport.getDateRangeOfAllStreams() != null) {
+            dateRange = new DateRange(studyStartDate, eventReport.getDateRangeOfAllStreams().getEndDate());
+        }
+        for (StudyReportWeek oneWeek : weeks) {
+            for (List<EventStreamDay> days : oneWeek.getByDayEntries().values()) {
+                for (EventStreamDay oneDay : days) {
+                    oneDay.setStudyBurstId(null);
+                    oneDay.setStudyBurstNum(null);
+                    oneDay.setSessionName(null);
+                    oneDay.setWeek(null);
+                    oneDay.setStartDay(null);
+                    for (EventStreamWindow window : oneDay.getTimeWindows()) {
+                        window.setEndDay(null);
+                        // This cannot be removed, or the window will be removed from persisted collection 
+                        // window.setTimeWindowGuid(null);
+                    }
+                }
+            }
+        }
+        StudyAdherenceReport report = new StudyAdherenceReport();
+        report.setDateRange(dateRange);
+        report.setWeeks(weeks);
+        report.setCurrentWeek(currentWeek);
+        report.setNextActivity(nextActivity);
+        report.setProgression(progression);
+        report.setAdherencePercent(adherence);
+        report.setEventTimestamps(eventTimestamps);
+        report.setUnsetEventIds(unsetEventIds);
+        report.setUnscheduledSessions(unscheduledSessions);
+        return report;
+    }
+
+    protected void calculateRowsAndLabels(StudyReportWeek oneWeek, LocalDate todayLocal) {
+        Set<String> labels = new LinkedHashSet<>();
+        Set<WeeklyAdherenceReportRow> rows = new LinkedHashSet<>();
+        for (List<EventStreamDay> days : oneWeek.getByDayEntries().values()) {
+            for (EventStreamDay oneDay : days) {
+                // The main searches to support are:
+                // <Study Burst>
+                // <Study Burst ID> 1
+                // <Study Burst ID> 1:Week 1
+                // <Session Name>
+                // <Session Name>:Week 1
+                // Week 1
+                String searchableLabel = (oneDay.getStudyBurstId() != null) ?
+                    String.format(":%s:%s %s:Week %s:%s:", oneDay.getStudyBurstId(), oneDay.getStudyBurstId(), oneDay.getStudyBurstNum(), oneWeek.getWeek(), oneDay.getSessionName()) :
+                    String.format(":%s:Week %s:", oneDay.getSessionName(), oneWeek.getWeek());
+                String displayLabel = (oneDay.getStudyBurstId() != null) ?
+                        String.format("%s %s / Week %s / %s", oneDay.getStudyBurstId(), oneDay.getStudyBurstNum(), oneWeek.getWeek(), oneDay.getSessionName()) :
+                        String.format("%s / Week %s", oneDay.getSessionName(), oneWeek.getWeek());
+                labels.add(searchableLabel);
```

Comment:
Check the real world search terms, like "Study Burst #3" which is in the label, but doesn't appear to pull up the record with one of these values.

Reply:
This works, but that doesn't guarantee that any label shown by a client app will work as is; in this case you'd have to look for ":Study Burst 3:" if you wanted an exact match for the 3rd study burst.